src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java

```diff
@@ -57,68 +58,47 @@ public BaseController(LookupDao inLookupDao, StreamingDao inStreamingDao, Naviga
 	}
 
 	protected void streamFlowLines(HttpServletResponse response,
-			String comid, String navigationMode, String stopComid, String distance, boolean legacy) throws IOException {
+			String comid, String navigationMode, String stopComid, String distance, boolean legacy) throws Exception {
 		Map<String, Object> parameterMap = parameters.processParameters(comid, navigationMode, distance, stopComid);
-		try (FlowLineTransformer transformer = new FlowLineTransformer(response)) {
-			if (legacy) {
-				String sessionId = getSessionId(parameterMap);
-				if (null != sessionId) {
-					parameterMap.put(StreamingDao.SESSION_ID, sessionId);
-					addContentHeader(response);
-					streamResults(transformer, BaseDao.FLOW_LINES_LEGACY, parameterMap);
-				} else {
-					response.setStatus(HttpStatus.BAD_REQUEST.value());
-				}
-			} else {
-				parameterMap.put(Parameters.COMID, NumberUtils.parseNumber(comid, Integer.class));
+		FlowLineTransformer transformer = new FlowLineTransformer(response);
+		if (legacy) {
+			String sessionId = getSessionId(parameterMap, response);
+			if (null != sessionId) {
+				parameterMap.put(StreamingDao.SESSION_ID, sessionId);
 				addContentHeader(response);
-				streamResults(transformer, BaseDao.FLOW_LINES, parameterMap);
+				streamResults(transformer, BaseDao.FLOW_LINES_LEGACY, parameterMap);
 			}
-
-		} catch (Exception e) {
-			LOG.error(e.getLocalizedMessage());
```

Comment:
This error handling method was exposing implementation details (things like sql errors & code) and preventing the global exception handling from working.